querier.go

```diff
@@ -221,6 +221,36 @@ func (r *postingsReader) Select(ms ...labels.Matcher) (Postings, []string) {
 	return p, absent
 }
 
+// findTuplesByPrefix uses binary search to find prefix matches within ts.
+func findTuplesByPrefix(m *labels.PrefixMatcher, ts StringTuples) (res []string, err error) {
```

Comment:
So named return types should be for godoc cleanliness: https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Having said that, I think `matches []string, err error` is more fitting. Also, just a `tuplesByPrefix` would be a better name IMO.